Extractor/RebrowseTriggerManager.cs

```diff
@@ -0,0 +1,151 @@
+                    {
+                        grouping.Add(nodeId, new List<ReferenceDescription>());
+                        namespaceNameToId.Add(refDef.DisplayName.ToString(), nodeId.ToString());
+                    }
+                    else if (
+                        grouping.ContainsKey(parent)
+                        // Ensures that the type of node being added is a variable node class
+                        && refDef.NodeClass == NodeClass.Variable
+                        // Filters targets nodes
+                        && targetNodes.Contains(refDef.DisplayName.ToString())
+                    )
+                    {
+                        grouping.GetValueOrDefault(parent).Add(refDef);
+                    }
+                },
+                token,
+                maxDepth: 1,
+                doFilter: false,
+                ignoreVisited: false
+            );
+
+            // To be used in filtering namespaces
+            var availableNamespaces = namespaceNameToId.Keys;
+
+            // Filters by namespaces
+            var processedNamespaces = (
+                shouldFilterNamespaces
+                    ? filteredNamespaces.Intersect(availableNamespaces)
+                    : availableNamespaces
+            ).ToList();
+
+            if (shouldFilterNamespaces && processedNamespaces.Count < filteredNamespacesCount)
+            {
+                logger.LogInformation(
+                    "Some namespaces were not found for rebrowse subscription as they do not exist on the server: {Namespaces}",
+                    filteredNamespaces.Except(processedNamespaces)
+                );
+            }
+
+            foreach (var @namespace in processedNamespaces)
+            {
+                var nodeId = namespaceNameToId.GetValueOrDefault(@namespace);
+                var references = grouping.GetValueOrDefault(nodeId);
+
+                nodeIds.AddRange(
+                    references.Where(@ref => targetNodes.Contains(@ref.DisplayName.ToString()))
+                        .Select(@ref => (NodeId)@ref.NodeId)
+                );
+            };
+
+            if (nodeIds.Count > 0) logger.LogInformation("The following nodes will be subscribed to a rebrowse: {nodes}", nodeIds);
+
+            var nodes = nodeIds.Select(node => new ServerItemSubscriptionState(_uaClient, node)).ToList();
+
+            await CreateSubscriptions(nodes, token);
+        }
+
+        private async Task CreateSubscriptions(List<ServerItemSubscriptionState> nodes, CancellationToken token)
+        {
+            var sub = await _uaClient.AddSubscriptions(
+                nodes, SubscriptionName,
+                (MonitoredItem item, MonitoredItemNotificationEventArgs _) =>
+                {
+                    var values = item.DequeueValues();
+                    var value = values.Count > 0 
+                        ? values[0].GetValue<System.DateTime>(UAExtractor.StartTime)
```

Comment:
Should we instead of using StartTime keep the time we last triggered a rebrowse, and default to the startup time? That might be a bit more consistent, in case we get multiple simultaneous notifications.